### DISCUSSION TITLE
Clean field containing URL

### DISCUSSION THREAD

**Discussion by gromit6891** [vectordotdev/vrl#1129] [Q&A]
How to effectively clean up a URL field that contains additional characters at the end, e.g "/MyWeb/Pages/favicon.ico -"?
I tried variations on:
.clean_url = parse_regex!("/MyWeb/Pages/favicon.ico -", r'(?P<url>.+?)\s-\s*$)')
But no luck so far.

**Comment by pront** [comment]
Hi @gromit6891,

Here is an alternative without regexes:
```
components = split!(.url, "/")
last_index = length(components) - 1
last = get!(components, [last_index])

i = find!(last, " ")
last = truncate!(last, i)

components = slice!(components, 0, last_index)
components = push(components, last)
.result = join!(components, "/")
```

If this answers your question, mark this as answered. 

---

I believe you can achieve what you want with a regex as well, maybe with: 
```
.clean_url = parse_regex!(.url_field, r'^(?P<clean_url>[^-\s]+(?:/[^-\s]+)*)\s*')
```
[playground link](https://playground.vrl.dev/?state=eyJwcm9ncmFtIjoiY29tcG9uZW50cyA9IHNwbGl0ISgudXJsLCBcIi9cIilcbmxhc3RfaW5kZXggPSBsZW5ndGgoY29tcG9uZW50cykgLSAxXG5sYXN0ID0gZ2V0IShjb21wb25lbnRzLCBbbGFzdF9pbmRleF0pXG5cbmkgPSBmaW5kIShsYXN0LCBcIiBcIilcbmxhc3QgPSB0cnVuY2F0ZSEobGFzdCwgaSlcblxuY29tcG9uZW50cyA9IHNsaWNlIShjb21wb25lbnRzLCAwLCBsYXN0X2luZGV4KVxuY29tcG9uZW50cyA9IHB1c2goY29tcG9uZW50cywgbGFzdClcbi5yZXN1bHQgPSBqb2luIShjb21wb25lbnRzLCBcIi9cIikiLCJldmVudCI6eyJ1cmwiOiIvTXlXZWIvUGFnZXMvZmF2aWNvbi5pY28gLSAtIC0gLSJ9LCJpc19qc29ubCI6ZmFsc2UsImVycm9yIjpudWxsfQ%3D%3D), but I didn't test if this is generically correct
